### PR TITLE
can configuration for rf transmission

### PR DESCRIPTION
**Pull request by MalwinaKozak**

### CODE REVIEW

File: additional data/ecumaster_can_frame_config.json [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+                "length": 8,
+                "factor": 0.0078125 ,
+                "offset": 0,
+                "signed": false,
+                "unit": "Lambda",
+                "byte_order": "little_endian"
+            },
+            "Lamcorr":
+            {
+                "name": "lambda",
```
Comment (cepekLP) [inline additional data/ecumaster_can_frame_config.json:180]: same name as previous

File: additional data/ecumaster_can_frame_config.json [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+            }
+        }
+    },
+    "ecumaster_3":
+    {
+        "id": "0x603",
+        "name": "ECUMASTER 3",
+        "signals":
+        {
+            "IGNANG":
```
Comment (cepekLP) [inline additional data/ecumaster_can_frame_config.json:145]: change name to ignition angle

File: additional data/ecumaster_can_frame_config.json [hunk 3]
```diff
@@ -0,0 +1,548 @@
...
+                "length": 8,
+                "factor": 0.5,
+                "offset": 0,
+                "signed": true,
+                "unit": "°",
+                "byte_order": "little_endian"
+            },
+            "DWELL":
+            {
+                "name": "DWELL",
```
Comment (cepekLP) [inline additional data/ecumaster_can_frame_config.json:158]: change name

File: additional data/ecumaster_can_frame_config.json [hunk 4]
```diff
@@ -0,0 +1,548 @@
...
+                "length": 8,
+                "factor": 1,
+                "offset": 0,
+                "signed": false,
+                "unit": "",
+                "byte_order": "little_endian"
+            },
+            "Fuel Pump State":
+            {
+                "name": "fps",
```
Comment (cepekLP) [inline additional data/ecumaster_can_frame_config.json:471]: change name to fuel_pump_state

File: code/data_filtration.py [hunk 5]
```diff
@@ -34,11 +39,27 @@ def init_kalman(self, lat, lon):
...
+    def calc_kilometers_driven(self):
+        radius_earth_km = 6371.0
+        if self.lat_prev is None or self.lon_prev is None:
+            self.lat_prev = self.lat_current
+            self.lon_prev = self.lon_current
+        delta_lat = self.lat_current - self.lat_prev
+        delta_lon = self.lon_current - self.lon_prev
+        a = math.sin(delta_lat / 2)**2 + math.cos(self.lat_prev) * math.cos(self.lat_current) * math.sin(delta_lon / 2)**2
+        c = 2 * math.atan2(math.sqrt(a), math.sqrt(1 - a))
+        distance_km = radius_earth_km * c
```
Comment (cepekLP) [inline code/data_filtration.py:61]: add function to calculate distance from longitude nad latitude delta

File: code/data_filtration.py [hunk 6]
```diff
@@ -73,23 +94,24 @@ def __init__(self) -> None:
...
-                self.f[i].x = np.array([[0.],
-                                        [self.gforce]])
-            
-            self.f[i].F = np.array([[1., 1/self.timestep], 
-                                    [0., 1.]])  # state transition matrix
-            self.f[i].H = np.array([[1., 0.]])  # Measurement function
-            self.f[i].P = np.array([[10., 0.], 
-                                    [0., 10.]])  # covariance matrix
-            self.f[i].R = np.array([[self.var_acc]])  # measurement noise
+            for i in range(3):
```
Comment (cepekLP) [inline code/data_filtration.py:97]: why to for functions?

File: code/func_abs.py [hunk 7]
```diff
@@ -75,20 +76,83 @@ def import_csv_abs(filepath, start_time):
...
+        .field(f"speed_{if_front}_right", speed_right)
+        .time(timestamp)
+    )
+    points.append(point)
+
+
+'''
+Live telemetry
+'''
+def vss_data_live(data_bytes, prev_speed, if_front):
```
Comment (cepekLP) [inline code/func_abs.py:120]: better would be is_front

File: code/func_abs.py [hunk 8]
```diff
@@ -75,20 +76,83 @@ def import_csv_abs(filepath, start_time):
...
+    idex 0: left
+    index 1: right
+
+    if_front: True if front, False if rear
+    
+    '''
+
+    match if_front:
+        case True:
+            if_front = 'front'
```
Comment (cepekLP) [inline code/func_abs.py:132]: better would be axle

File: code/func_abs.py [hunk 9]
```diff
@@ -75,20 +76,83 @@ def import_csv_abs(filepath, start_time):
...
+        case False:
+            if_front = 'rear'
+
+    flow_multiplier = 32.
```
Comment (cepekLP) [inline code/func_abs.py:136]: flow?

File: code/func_ecumaster.py [hunk 10]
```diff
@@ -324,46 +399,166 @@ def engine_data2(data, timestamp, points):
...
         .tag("engine", 'raw')
-        .field("Coolant fan", coolant_fan)
+        .field("coolant_fan", coolant_fan)
         .time(timestamp)
     )
     points.append(point)
 
+
+'''
+LIve telemetry.
```
Comment (cepekLP) [inline code/func_ecumaster.py:440]:
```suggestion
Live telemetry.
```

**Review by cepekLP — CHANGES_REQUESTED** [review]

**Review by cepekLP — CHANGES_REQUESTED** [review]

File: code/func_gps.py [hunk 11]
```diff
@@ -204,4 +205,241 @@ def convert_csv_gps(filepath):
...
+        self.speed_one_lap_container = []
+        self.lat_prev = None
+        self.lon_prev = None
+        self.lat_current = None
+        self.lon_current = None
+
+    def callback(self, frame_data):
+        lat = frame_data['latitude']
+        lon = frame_data['longitude']
+        lat = (float(lat) / 100 // 1) + (float(lat) % 100.0 / 60.0)
```
Comment (cepekLP) [inline code/func_gps.py:370]: create function to calc latitude and longitude

File: code/func_gps.py [hunk 12]
```diff
@@ -204,4 +205,241 @@ def convert_csv_gps(filepath):
...
+            self.lon_current = self.f[1].x[0][0]
+
+    def calc_kilometers_driven(self):
+        radius_earth_km = 6371.0
+        if self.lat_prev is None or self.lon_prev is None:
+            self.lat_prev = self.lat_current
+            self.lon_prev = self.lon_current
+        delta_lat = self.lat_current - self.lat_prev
+        delta_lon = self.lon_current - self.lon_prev
+        a = math.sin(delta_lat / 2)**2 + math.cos(self.lat_prev) * math.cos(self.lat_current) * math.sin(delta_lon / 2)**2
```
Comment (cepekLP) [inline code/func_gps.py:423]: create function to calc distance from delta lat and long

File: code/func_gps.py [hunk 13]
```diff
@@ -204,4 +205,241 @@ def convert_csv_gps(filepath):
...
+    #     data_to_send["best_lap_number"] = best_lap_number
+
+    # if lap_diff != 0:
+    #     data_to_send["lap_time"] = last_lap_time
+    #     data_to_send["lap_counter"] = lap_counter
+    #     #data_to_send["lap_duration_time"] = lap_duration_time
+
+    return data_to_send
+
+def gps_additional_data_live(data_bytes):
```
Comment (cepekLP) [inline code/func_gps.py:335]: remove unused function

File: code/func_gps.py [hunk 14]
```diff
@@ -204,4 +205,241 @@ def convert_csv_gps(filepath):
...
+        .field("longitude", lon)
+        .time(timestamp)
+    )
+    points.append(point)
+
+'''
+live telemetry.
+'''
+
+def gps_data_live(data_bytes, filter, lap_timer):
```
Comment (cepekLP) [inline code/func_gps.py:288]: remove unused function

File: code/lap_timer.py [hunk 15]
```diff
@@ -2,35 +2,27 @@
...
 
 
 class LapTimer:
     def __init__(
             self,
-            x1: float = 18.713115,
-            y1: float = 54.178896,
-            x2: float = 18.713045,
-            y2: float = 54.178709,
+            x1: float = 14.7616,
```
Comment (cepekLP) [inline code/lap_timer.py:12]: we need to prepare some config file with those values

File: additional data/ecumaster_can_frame_config.json [hunk 16]
```diff
@@ -0,0 +1,567 @@
...
+                "length": 8,
+                "factor": 1,
+                "offset": 0,
+                "signed": false,
+                "unit": "",
+                "byte_order": "little_endian"
+            },
+            "Can Switch 1":
+            {
+                "name": "cansw1",
```
Comment (cepekLP) [inline additional data/ecumaster_can_frame_config.json:391]: cahnge to can_sw1

**Review by cepekLP — CHANGES_REQUESTED** [review]

File: code/data_filtration.py (hunk 5 above)
Comment (MalwinaKozak) [reply code/data_filtration.py:61]: class moved to func_gps.py

File: additional data/ecumaster_can_frame_config.json [hunk 17]
```diff
@@ -0,0 +1,567 @@
...
+        {
+            "Ignition angle":
+            {
+                "name": "ignition_angle",
+                "start_bit": 0,
+                "length": 8,
+                "factor": 0.5,
+                "offset": 0,
+                "signed": true,
+                "unit": "°",
```
Comment (cepekLP) [inline additional data/ecumaster_can_frame_config.json:172]: change to deg / degrees

File: additional data/ecumaster_can_frame_config.json [hunk 18]
```diff
@@ -0,0 +1,567 @@
...
+            },
+            "Fuel pressure":
+            {
+                "name": "fuel_pressure",
+                "start_bit": 40,
+                "length": 8,
+                "factor": 0.0625,
+                "offset": 0,
+                "signed": false,
+                "unit": "Bar",
```
Comment (cepekLP) [inline additional data/ecumaster_can_frame_config.json:142]:
```suggestion
                "unit": "bar",
```

File: additional data/ecumaster_can_frame_config.json [hunk 19]
```diff
@@ -0,0 +1,567 @@
...
+            },
+            "Oil pressure":
+            {
+                "name": "oil_pressure",
+                "start_bit": 32,
+                "length": 8,
+                "factor": 0.0625,
+                "offset": 0,
+                "signed": false,
+                "unit": "Bar",
```
Comment (cepekLP) [inline additional data/ecumaster_can_frame_config.json:131]:
```suggestion
                "unit": "bar",
```

File: additional data/ecumaster_can_frame_config.json (hunk 17 above)
Comment (cepekLP) [inline additional data/ecumaster_can_frame_config.json:172]:
```suggestion
                "unit": "deg",
```

File: additional data/telemetry_can_frame_config.json [hunk 20]
```diff
@@ -0,0 +1,1015 @@
...
+            },
+            "Burned Fuel":
+            {
+                "name": "burned_fuel",
+                "start_bit": 32,
+                "length": 16,
+                "factor": 0.00390625,
+                "offset": 0,
+                "signed": false,
+                "unit": "L",
```
Comment (cepekLP) [inline additional data/telemetry_can_frame_config.json:38]:
```suggestion
                "unit": "l",
```

File: additional data/telemetry_can_frame_config.json [hunk 21]
```diff
@@ -0,0 +1,1015 @@
...
+            },
+            "Speed":
+            {
+                "name": "speed_gps",
+                "start_bit": 16,
+                "length": 16,
+                "factor": 0.0625,
+                "offset": 0,
+                "signed": false,
+                "unit": "",
```
Comment (cepekLP) [inline additional data/telemetry_can_frame_config.json:950]:
```suggestion
                "unit": "km/h",
```

File: additional data/telemetry_can_frame_config.json [hunk 22]
```diff
@@ -0,0 +1,1015 @@
...
+            },
+            "Course":
+            {
+                "name": "course",
+                "start_bit": 32,
+                "length": 16,
+                "factor": 0.5,
+                "offset": 0,
+                "signed": false,
+                "unit": "",
```
Comment (cepekLP) [inline additional data/telemetry_can_frame_config.json:961]:
```suggestion
                "unit": "deg",
```

File: additional data/telemetry_can_frame_config.json [hunk 23]
```diff
@@ -0,0 +1,1015 @@
...
+        {
+            "Altitude":
+            {
+                "name": "altitude",
+                "start_bit": 0,
+                "length": 16,
+                "factor": 0.125,
+                "offset": 0,
+                "signed": false,
+                "unit": "",
```
Comment (cepekLP) [inline additional data/telemetry_can_frame_config.json:939]:
```suggestion
                "unit": "m",
```

File: additional data/telemetry_can_frame_config.json [hunk 24]
```diff
@@ -0,0 +1,1015 @@
...
+        {
+            "Rear":
+            {
+                "name": "rear_brake_pressure",
+                "start_bit": 0,
+                "length": 16,
+                "factor": 0.015625,
+                "offset": 0,
+                "signed": false,
+                "unit": "B",
```
Comment (cepekLP) [inline additional data/telemetry_can_frame_config.json:980]:
```suggestion
                "unit": "bar",
```

File: additional data/telemetry_can_frame_config.json [hunk 25]
```diff
@@ -0,0 +1,1015 @@
...
+            },
+            "Front":
+            {
+                "name": "front_brake_pressure",
+                "start_bit": 16,
+                "length": 16,
+                "factor": 0.015625,
+                "offset": 0,
+                "signed": false,
+                "unit": "B",
```
Comment (cepekLP) [inline additional data/telemetry_can_frame_config.json:991]:
```suggestion
                "unit": "bar",
```

**Review by cepekLP — APPROVED** [review]
🥇

**Pull request merged**